website/static/js/fangorn.js

```diff
@@ -49,6 +49,64 @@ var getExtensionIconClass = function(name) {
     return null;
 };
 
+function findByTempID(parent, tmpID){
+    var child;
+    var item;
+    for (var i = 0; i < parent.children.length; i++) {
+        child = parent.children[i];
+        if (!child.data.tmpID) {
+            continue;
+        }
+        if (child.data.tmpID === tmpID) {
+            item = child;
+        }
+    }
+    return item;
+}
+
+function cancelUploads (row) {
+    var treebeard = this;
+    var filesArr = treebeard.dropzone.getQueuedFiles();
+    for (var i = 0; i < filesArr.length; i++) {
+        var m = filesArr[i];
+        if(!row){
+            var parent = m.treebeardParent || treebeard.dropzoneItemCache;
+            var item = findByTempID(parent, m.tmpID);
+            treebeard.dropzone.removeFile(m);
+            treebeard.deleteNode(parent.id,item.id);
+        } else {
+            treebeard.deleteNode(row.parentID,row.id);
```

Comment:
I don't think that's right at this point in the code. If we get to the `else` block, then `row` is truthy, which means we're supposed to be canceling only one upload and deleting only one row--I think. Am I reading this correctly?